cylc/flow/etc/job.sh

```diff
@@ -187,6 +199,20 @@ cylc__set_return() {
     return "${1:-0}"
 }
 
+###############################################################################
+# Save the data using cylc message and exit the profiler
+cylc__kill_profiler() {
+    if [[ -f "profiler.json" ]]; then
+      max_rss="$(jq -r '.max_rss' profiler.json)"
+      cpu_time="$(jq -r '.cpu_time' profiler.json)"
+      cylc message -- "${CYLC_WORKFLOW_ID}" "${CYLC_TASK_JOB}" "DEBUG: cpu_time $cpu_time max_rss $max_rss" || true
+      rm profiler.json
+    fi
+    if [[ -f "proc/${profiler_pid}" ]]; then
+      kill -s SIGINT "${profiler_pid}" || true
+    fi
+}
```

Comment:
Neat trick, but unfortunately, the `jq` command is not part of the POSIX standard so we cannot rely on it being installed.

The easiest solution would be to send the full JSON text as the message, the only catch is to make sure you strip any newline characters (note the [`tr` command is POSIX](https://www.man7.org/linux/man-pages/man1/tr.1p.html)):

```suggestion
cylc__kill_profiler() {
    if [[ -f "profiler.json" ]]; then
      cylc message -- "${CYLC_WORKFLOW_ID}" "${CYLC_TASK_JOB}" "DEBUG: $(cat profiler.json | tr -d '\n')" || true
      rm profiler.json
    fi
    if [[ -f "proc/${profiler_pid}" ]]; then
      kill -s SIGINT "${profiler_pid}" || true
    fi
}
```

The UIS end of this PR would need to adapt to the change from `cpu_time <value>` to `"cpu_time": <value>` if going down this route.

Reply:
Done